uint8 pin
string frame
uint32 timeout
---
uint8 error
